### PR TITLE
Add SpawnIfUnobstructed IEntityManager extension method

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
Also adds TrySpawnIfUnobstructed. It takes in a Box2 and CollisionGroup to check for.
Tests for both methods have been added.

### CODE REVIEW

File: Content.Shared/Utility/EntitySystemExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+                if (collisionLayer == 0 || (physics.CollisionMask & (int) collisionLayer) == 0)
+                {
+                    continue;
+                }
+
+                return null;
+            }
+
+            return entityManager.SpawnEntity(prototypeName, coordinates);
```
Comment (metalgearsloth) [inline Content.Shared/Utility/EntitySystemExtensions.cs:54]: Should use GetCollidingEntities directly if you're getting the physics components.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
n u l l a b l e s

File: Content.Shared/Utility/EntitySystemExtensions.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Shared/Utility/EntitySystemExtensions.cs]: Done. Also the nullables are in, just not for tests as those are really annoying with nullable and very few use it. Should I enable it for the test as well?

Comment (DrSmugleaf) [reply Content.Shared/Utility/EntitySystemExtensions.cs]: Done for the test as well.

**Review by PaulRitter — APPROVED** [review]

**Pull request merged**